### PR TITLE
feat: mapping token for /quoterequests and /transfers

### PR DESCRIPTION
**Pull request by elijah0kello**
## This PR implements the following at a high level
- Implements token mapping for /quoterequests
- Implements token mapping for /transfers
- creates unit tests for the above functionality
- implemements ttk tests for the functions

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -170,19 +170,32 @@ jobs:
           command: npm run dep:check
 
   test-unit:
```
Comment (vijayg10) [inline .circleci/config.yml]: If you are running dependencies, these are not unit tests anymore. You can put them in integration tests or functional tests pipeline.

File: README.md [hunk 2]
```diff
@@ -48,10 +48,23 @@ npm run start
 ```
 
 # Test
+Before you run unit tests, make sure to run the ttk backend using this command.
```
Comment (vijayg10) [inline README.md]: Not unit tests. I think these are functional tests.

File: src/domain/interfaces/types.ts [hunk 3]
```diff
@@ -36,8 +36,89 @@ export interface IPaymentTokenMapping {
...
+    idValue: string,
+    lastName: string,
+    merchantClassificationCode: string,
+    middleName: string,
+    type: string,
+    supportedCurrencies: string,
+    kycInformation: string
+}
+
+export type IQuoteRequestData = {
```
Comment (vijayg10) [inline src/domain/interfaces/types.ts]: These types are exported in api-snippets library. If you want you can directly use those types from there.

File: src/token-adapter-svc/sdkRoutes.ts [hunk 4]
```diff
@@ -74,4 +90,44 @@ export class SDKRoutes implements IRoutes {
...
+
+        const payload: IQuoteRequestData = request.payload as IQuoteRequestData;
+
+        if(!payload.to){
+            return h.response("Bad Request: Payload missing crucial info").code(400);
+        }
+
+        const result = await this.sdkAggregate.postQuotes(payload);
+        if(!result){
+            return h.response({statusCode:3204,message:"Party not found"}).code(404);
```
Comment (vijayg10) [inline src/token-adapter-svc/sdkRoutes.ts]: I think this error message should be more specific. Like alias not found or matched...etc

File: src/api-spec/payment-token-adapter-spec.yaml [hunk 5]
```diff
@@ -58,6 +60,523 @@ paths:
...
+                    items:
+                      type: string
+                  kycInformation:
+                    type: string
+
+          description: 'Response containing details of the requested party'
+      tags:
+        - Discovery
+
+  /quoterequests:
```
Comment (vijayg10) [inline src/api-spec/payment-token-adapter-spec.yaml]: I think this swagger should be split into two. One for connector service and another for token service.

Comment (elijah0kello) [reply src/api-spec/payment-token-adapter-spec.yaml]: The issue is some of these routes are shared by SDK Scheme Adapter, Core connector and the Payment Token Adapter

File: src/domain/interfaces/types.ts (hunk 3 above)
Comment (elijah0kello) [reply src/domain/interfaces/types.ts]: Let me test this out.

**Comment by elijah0kello** [comment]
I have pushed commits to address some of the issues raised.

**Review by vijayg10 — APPROVED** [review]

**Pull request merged**